{
  "totalNumberSkippingTests": 17,
  "bugType": "passing_passing",
  "totalNumberErroringTests": 0,
  "repo": "google/error-prone",
  "metrics": {
    "FailureNames": [
      "java.lang.AssertionError",
      "skipped"
    ],
    "StepsDurationsInSeconds": {
      "CheckoutBuggyBuildSourceCode": 0,
      "ComputeClasspath": 1,
      "ComputeTestDir": 0,
      "GatherTestInformationPreviousBuildSourceCode": 0,
      "ResolveDependency": 11,
      "BuildProjectBuild": 15,
      "BuildProjectPreviousBuildSourceCode": 26,
      "TestProjectBuild": 658,
      "ComputeSourceDir": 0,
      "InitRepoToPush": 5,
      "TestProjectPreviousBuildSourceCode": 624,
      "GatherTestInformationBuild": 0,
      "CloneRepository": 52,
      "CheckoutPatchedBuild": 0,
      "NopolRepair": 40,
      "PushIncriminatedBuild": 4,
      "CommitPatch": 5
    },
    "BugCommitUrl": "http://github.com/google/error-prone/commit/e049633444b3b35680762e8f03fa4ed34804deab",
    "NbFailingTests": 4,
    "PatchCommitUrl": "http://github.com/google/error-prone/commit/3974a3cccc61bb7d80a5c4090f2efcbc3e870b42",
    "ReproductionDate": "May 18, 2017 10:44:41 AM",
    "BuggyBuildId": 193176794,
    "PatchedBuilId": 193900872,
    "BuggyBuildURL": "http://travis-ci.org/google/error-prone/builds/193176794",
    "PatchedBuildURL": "http://travis-ci.org/google/error-prone/builds/193900872",
    "BuggyBuildDate": "Jan 18, 2017 9:42:41 PM",
    "PatchedBuildDate": "Jan 21, 2017 1:55:06 AM",
    "BugCommit": "e049633444b3b35680762e8f03fa4ed34804deab",
    "PatchCommit": "3974a3cccc61bb7d80a5c4090f2efcbc3e870b42",
    "PatchChangedFiles": 6,
    "PatchAddedLines": 0,
    "PatchDeletedLines": 0,
    "NbRunningTests": 1855,
    "NbLibraries": 52,
    "NbFileApp": 1018,
    "NbFileTests": 1400,
    "NbCPU": 56,
    "FreeMemory": 2556306512,
    "TotalMemory": 4756865024,
    "AngelicValuesByTest": {
      "com/google/errorprone/bugpatterns/threadsafety/GuardedByCheckerTest": 0,
      "com/google/errorprone/bugpatterns/FinallyTest": 0,
      "com/google/errorprone/bugpatterns/threadsafety/ImmutableEnumCheckerTest": 0
    },
    "FreeMemoryByStep": {
      "CheckoutBuggyBuildSourceCode": 1181292008,
      "ComputeClasspath": 1494982984,
      "ComputeTestDir": 1221462856,
      "GatherTestInformationPreviousBuildSourceCode": 1052970080,
      "ResolveDependency": 1162924072,
      "BuildProjectBuild": 3524311424,
      "BuildProjectPreviousBuildSourceCode": 1145594104,
      "TestProjectBuild": 3498659344,
      "ComputeSourceDir": 1234849176,
      "InitRepoToPush": 1499505864,
      "TestProjectPreviousBuildSourceCode": 1093605328,
      "GatherTestInformationBuild": 3466179344,
      "CloneRepository": 1292335992,
      "CheckoutPatchedBuild": 3547753504,
      "NopolRepair": 3647409168,
      "PushIncriminatedBuild": 2782678504,
      "CommitPatch": 2556306512
    }
  },
  "bugCommit": "https://github.com/google/error-prone/compare/cb7eb6183143...e049633444b3",
  "failingModule": "/root/workspace/google/error-prone/193176794/core",
  "hostname": "spirals-spoon3r",
  "totalNumberFailingTests": 4,
  "error-types": [
    "java.lang.AssertionError",
    "skipped"
  ],
  "totalNumberRunningTests": 1855,
  "failing-test-cases": [
    {
      "className": "com.google.errorprone.bugpatterns.FinallyTest",
      "failingMethods": [
        "com.google.errorprone.bugpatterns.FinallyTest#lambda"
      ],
      "erroringMethods": [],
      "failures": [
        {
          "failureName": "java.lang.AssertionError",
          "failureDetail": "Saw unexpected error on line 6. All errors:\n[/Test.java:6: warning: [Finally] If you return or throw from a finally, then values returned or thrown from the try-catch block will be ignored. Consider using try-with-resources instead.\n      Runnable r \u003d () -\u003e { return; };\n                           ^\n    (see http://errorprone.info/bugpattern/Finally)]",
          "isError": false
        }
      ],
      "nbFailures": 1,
      "nbErrors": 0
    },
    {
      "className": "com.google.errorprone.bugpatterns.threadsafety.GuardedByCheckerTest",
      "failingMethods": [
        "com.google.errorprone.bugpatterns.threadsafety.GuardedByCheckerTest#qualifiedMethod",
        "com.google.errorprone.bugpatterns.threadsafety.GuardedByCheckerTest#innerClassTypeQualifier",
        "com.google.errorprone.bugpatterns.threadsafety.GuardedByCheckerTest#testReadWriteLockCopy",
        "com.google.errorprone.bugpatterns.threadsafety.GuardedByCheckerTest#tryWithResources",
        "com.google.errorprone.bugpatterns.threadsafety.GuardedByCheckerTest#qualfiedType",
        "com.google.errorprone.bugpatterns.threadsafety.GuardedByCheckerTest#lambda"
      ],
      "erroringMethods": [],
      "failures": [
        {
          "failureName": "java.lang.AssertionError",
          "failureDetail": "Did not see an error on line 7 matching should be guarded by \u0027this\u0027,. All errors:\n[]",
          "isError": false
        },
        {
          "failureName": "skipped",
          "failureDetail": "skipped",
          "isError": false
        },
        {
          "failureName": "skipped",
          "failureDetail": "skipped",
          "isError": false
        },
        {
          "failureName": "skipped",
          "failureDetail": "skipped",
          "isError": false
        },
        {
          "failureName": "skipped",
          "failureDetail": "b/26834754",
          "isError": false
        },
        {
          "failureName": "skipped",
          "failureDetail": "b/26834754",
          "isError": false
        }
      ],
      "nbFailures": 6,
      "nbErrors": 0
    },
    {
      "className": "com.google.errorprone.bugpatterns.threadsafety.ImmutableEnumCheckerTest",
      "failingMethods": [
        "com.google.errorprone.bugpatterns.threadsafety.ImmutableEnumCheckerTest#finalMutableField",
        "com.google.errorprone.bugpatterns.threadsafety.ImmutableEnumCheckerTest#annotatedEnumThatImplementsImmutableInterface"
      ],
      "erroringMethods": [],
      "failures": [
        {
          "failureName": "java.lang.AssertionError",
          "failureDetail": "Saw unexpected error on line 2. All errors:\n[/Test.java:2: warning: [ImmutableEnumChecker] enums are immutable by default; annotating them with @Immutable is unnecessary\n@Immutable\n^\n    (see http://errorprone.info/bugpattern/ImmutableEnumChecker)\n  Did you mean to remove this line?]",
          "isError": false
        },
        {
          "failureName": "java.lang.AssertionError",
          "failureDetail": "Did not see an error on line 7 matching enums should only have immutable fields, \u0027Set\u0027 is mutable. All errors:\n[/Test.java:7: warning: [ImmutableEnumChecker] enums should be immutable, \u0027Set\u0027 is mutable\n  final Set\u003cInteger\u003e xs;\n                     ^\n    (see http://errorprone.info/bugpattern/ImmutableEnumChecker)]",
          "isError": false
        }
      ],
      "nbFailures": 2,
      "nbErrors": 0
    }
  ]
}